### PR TITLE
Add pystem

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by NingWang1990** [comment]
Start...

**Comment by jan-janssen** [comment]
@NingWang1990 Is it ok if I add you as a maintainer? 

Just to follow the rules of conda-forge: 

> GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

**Comment by NingWang1990** [comment]
sure, the maintainer is fine for me.

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

File: recipes/pystem/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - pip
+    - python
+    - fftw
+    - numpy >=1.17.0
```
Comment (chrisburr) [inline recipes/pystem/meta.yaml:25]: This is [using the numpy C API](https://github.com/NingWang1990/pySTEM/blob/0f2de23d790d9723dd6175663282279843c0a73a/pystem/stemdescriptor2.cpp#L7) so you need to follow:

https://conda-forge.org/docs/maintainer/knowledge_base.html#linking-numpy

I'd also encourage you to drop the version pin and use the global numpy pinning for the reasons stated in https://github.com/conda-forge/staged-recipes/pull/12043#discussion_r448363865

File: recipes/pystem/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  run:
+    - python
+    - _openmp_mutex  # [linux]
```
Comment (chrisburr) [inline recipes/pystem/meta.yaml:28]: https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp

**Comment by jan-janssen** [comment]
@chrisburr Thanks for your feedback, I updated the recipe accordingly.

**Pull request merged**